source/common/tcp_proxy/tcp_proxy.h

```diff
@@ -135,6 +136,19 @@ class Config {
 
 typedef std::shared_ptr<Config> ConfigSharedPtr;
 
+/**
+ * Per-connection TCP Proxy Filter configuration.
+ */
+class PerConnectionTcpProxyConfig : public Envoy::RequestInfo::FilterState::Object {
```

Comment:
I'd name this PerConnectionCluster. It's already in namespace TcpProxy, and we don't want this to grow to include more stuff. If there's more stuff to set on TcpProxy it should be set as separate per-connection values, so that it doesn't all have to be set by the same filter.

Reply:
But can't filters retrieve and modify the same object? Having the settings in the same object does not preclude multiple filters from using it does it?

Reply:
It makes the interface a little stranger. Instead of just setting a thing, you have to test if it exists first, and if it does take one path, else create a new one and insert it.

Reply:
Fair point, but this is based on the assumption that multiple filters will be setting independent things. Now, even in that scenario, given that any other filter could have set the variable before mine, to prevent an exception, I would have to do the check-if-it-exists and then set the thing. [The interface throws an error if I try to set an object that already exists].

Make sense? the check before update will be part of the normal pattern in a scenario where there could be multiple filters. Given that, I feel having a standard check if the umbrella object exists and then do setters/getters is cleaner than check if my object exists and then update..

I don't have strong feelings about this as I don't have data to support either side. So, if you feel like per conn cluster is better, I will happily change it.. We can always consolidate if need be